### PR TITLE
Fix column reordering when mouse leaves the browser window.

### PR DESCRIPTION
**Pull request by sisyphean2**
## Description
Add an onMouseLeave function that handles the mouse leaving the browser window. This function closes all listener functionality related to column reordering and cancels the reorder.

## Motivation and Context
https://github.com/schrodinger/fixed-data-table-2/issues/104

## How Has This Been Tested?
I tested this manually by reordering columns multiple times, dropping the columns both inside and outside the browser window.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x]  My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: src/vendor_upstream/dom/DOMMouseMoveTracker.js [hunk 1]
```diff
@@ -61,6 +62,16 @@ class DOMMouseMoveTracker {
...
       );
+      this._eventLeaveToken = EventListener.listen(
+        this._domNode,
+        'mouseleave',
+        this._onMouseLeave
+      );
+      this._eventOutToken = EventListener.listen(
+        this._domNode,
+        'mouseout',
+        this.onMouseLeave
```
Comment (KamranAsif) [inline src/vendor_upstream/dom/DOMMouseMoveTracker.js]: Maybe rename onMouseLeave to onMouseEnd?

File: src/vendor_upstream/dom/DOMMouseMoveTracker.js [hunk 2]
```diff
@@ -141,6 +156,13 @@ class DOMMouseMoveTracker {
     }
     this._onMoveEnd();
```
Comment (KamranAsif) [inline src/vendor_upstream/dom/DOMMouseMoveTracker.js]: Lets explicitly return false here

File: src/vendor_upstream/dom/DOMMouseMoveTracker.js (hunk 1 above)
Comment (sisyphean2) [inline src/vendor_upstream/dom/DOMMouseMoveTracker.js]: Sure!

File: src/vendor_upstream/dom/DOMMouseMoveTracker.js (hunk 2 above)
Comment (sisyphean2) [inline src/vendor_upstream/dom/DOMMouseMoveTracker.js]: OK.

File: src/vendor_upstream/dom/DOMMouseMoveTracker.js [hunk 3]
```diff
@@ -77,11 +88,15 @@ class DOMMouseMoveTracker {
    * These releases all of the listeners on document.body.
    */
   releaseMouseMoves() {
-    if (this._eventMoveToken && this._eventUpToken) {
+    if (this._eventMoveToken && this._eventUpToken && this._eventLeaveToken) {
```
Comment (sisyphean2) [inline src/vendor_upstream/dom/DOMMouseMoveTracker.js]: I need to add this.eventOutToken to this list.

File: src/FixedDataTable.js [hunk 4]
```diff
@@ -773,14 +773,17 @@ var FixedDataTable = React.createClass({
...
-    this.props.onColumnReorderEndCallback({
-      columnBefore, columnAfter, reorderColumn
-    });
+    if (!cancelReorder) {
+      this.props.onColumnReorderEndCallback({
+        columnBefore, columnAfter, reorderColumn
+      });
+    }
 
     var onHorizontalScroll = this.props.onHorizontalScroll;
```
Comment (KamranAsif) [inline src/FixedDataTable.js]: I don't think we want to run this onHorizontalScroll code here.
Can we just return early if cancelReorder is true?

**Comment by KamranAsif** [comment]
LGTM after one comment is addressed

File: src/FixedDataTable.js (hunk 4 above)
Comment (sisyphean2) [inline src/FixedDataTable.js]: Good call. I only made sure that we want the stuff above the cancelReorder check, I didn't look below it to see if we want to skip anything.

I'll change the contents of the if block to return early and put the Callback after the if block.

**Pull request merged**